odxtools/diaglayer.py

```diff
@@ -461,16 +462,16 @@ def decode_response(
 
     def get_communication_parameter(self,
                                     name: str,
-                                    functional: Optional[bool] = None,
+                                    is_functional: Optional[bool] = None,
                                     protocol_name: Optional[str] = None) \
             -> Optional[CommunicationParameterRef]:
 
         cps = [cp for cp in self.communication_parameters if cp.short_name == name]
 
-        if functional is not None:
-            cps = [cp for cp in cps if cp.is_functional == functional]
+        if is_functional is not None:
+            cps = [cp for cp in cps if cp.is_functional == is_functional]
```

Comment:
```suggestion
            cps = [cp for cp in cps if cp.is_functional in (None, is_functional)]
```

(at least if the comparam does not need to specify the "FUNCTIONAL" attribute and if its absence means "don't care"...)

Reply:
The way the `is_functional` attribute is set for a instance of type `CommunicationParameterRef`, it is currently not possible to have a `None` value at this attribute:

`is_functional = (dl_type == DIAG_LAYER_TYPE.FUNCTIONAL_GROUP)`

... except someone initializes these `CommunicationParameterRef` objects in the future in a different way.

Reply:
fair enough. thanks for the explaination